pkgs/by-name/as/async-profiler/package.nix

```
@@ -0,0 +1,42 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  jdk,
+  makeWrapper,
+}:
+stdenv.mkDerivation rec {
+  pname = "async-profiler";
+  version = "3.0";
+
+  src = fetchFromGitHub {
+    owner = "jvm-profiling-tools";
+    repo = "async-profiler";
+    rev = "v${version}";
+    hash = "sha256-0CCJoRjRLq4LpiRD0ibzK8So9qSQymePCTYUI60Oy2k=";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  buildInputs = [ jdk ];
+
+  installPhase = ''
+    runHook preInstall
+    install -D build/bin/asprof "$out/bin/async-profiler"
+    install -D build/lib/libasyncProfiler.so "$out/lib/libasyncProfiler.so"
```

```suggestion
    install -D build/lib/libasyncProfiler${stdenv.hostPlatform.extensions.sharedLibrary} -t "$out/lib/libasyncProfiler"
```

Can't test since I'm on a long distance train with poor connectivity.